spec/defines/system_hook_spec.rb

```diff
@@ -0,0 +1,59 @@
+require 'spec_helper'
+
+describe 'gitlab::system_hook' do
+  let(:title) { 'test-system-hook' }
+
+  let(:pre_condition) do
+    <<-MANIFEST
+    class { 'gitlab':
+      repository_configuration => {},
+    }
+    MANIFEST
+  end
+
+  context "with type => #{type} and source" do
+    let(:source) { 'puppet:///modules/my_module/file-hook' }
+    let(:params) do
+      {
+        custom_hooks_dir: '/custom/hooks/dir',
+        source: source
+      }
+    end
+
+    it { is_expected.to compile }
```

Comment:
please ensure that the whole catalog compiles:
```suggestion
    it { is_expected.to compile.with_all_deps }
```